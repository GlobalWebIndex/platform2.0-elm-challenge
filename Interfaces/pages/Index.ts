import { Breeds } from "./Breeds";

export interface Cat {
    id: string,
    url: string,
    width: number,
    height: number
};

export type CatList = Cat[];

export interface CatImage extends Cat {
    categories: [
        {
            id: number,
            name: string
        }
    ],
    breeds: Breeds[]
};