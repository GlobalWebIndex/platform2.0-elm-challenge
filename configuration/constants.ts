export const constants = {
    sub_id: "dkfgnerifw39iefj",
    apikey: "live_RrdwcXm3saxuBPOgkxbfd3sbSfELZUhKFo5RhCwKlCNvj5FfDKMGHqH3z0pqcfp1",
    order: "RANDOM",
    limit: 10,
    size: "thumb"
};