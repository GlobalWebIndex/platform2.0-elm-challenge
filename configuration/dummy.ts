//menu static data
export const menuData = {
    items: [
        {
            id: 1,
            href: "/",
            label: "home"
        },
        {
            id: 2,
            href: "/breeds",
            label: "breeds"
        },
        {
            id: 3,
            href: "/favorites",
            label: "favorites"
        }
    ]
};

//initialize context
export const initialContext = {
    loading: false,
    darkMode: false,
    favorites: [],
    addToFavorites: () => { },
    removeFromFavorites: () => { },
    setLoading: () => { },
    setDarkMode: () => { }
};

//index page dummy contert
export const dummyIndex = [
    {
        "id": "7gn",
        "url": "https://cdn2.thecatapi.com/images/7gn.jpg",
        "width": 387,
        "height": 592
    },
    {
        "id": "9sp",
        "url": "https://cdn2.thecatapi.com/images/9sp.jpg",
        "width": 1024,
        "height": 768
    },
    {
        "id": "a8k",
        "url": "https://cdn2.thecatapi.com/images/a8k.jpg",
        "width": 500,
        "height": 750
    },
    {
        "id": "af8",
        "url": "https://cdn2.thecatapi.com/images/af8.jpg",
        "width": 500,
        "height": 333
    },
    {
        "id": "b0n",
        "url": "https://cdn2.thecatapi.com/images/b0n.jpg",
        "width": 341,
        "height": 500
    },
    {
        "id": "bkc",
        "url": "https://cdn2.thecatapi.com/images/bkc.jpg",
        "width": 400,
        "height": 300
    },
    {
        "id": "cfk",
        "url": "https://cdn2.thecatapi.com/images/cfk.jpg",
        "width": 553,
        "height": 400
    },
    {
        "id": "MTUwNjcwMQ",
        "url": "https://cdn2.thecatapi.com/images/MTUwNjcwMQ.jpg",
        "width": 1200,
        "height": 800
    },
    {
        "id": "NVGe5-dNr",
        "url": "https://cdn2.thecatapi.com/images/NVGe5-dNr.jpg",
        "width": 4032,
        "height": 3024
    },
    {
        "id": "Z6UtarTTe",
        "url": "https://cdn2.thecatapi.com/images/Z6UtarTTe.png",
        "width": 200,
        "height": 200
    }
];