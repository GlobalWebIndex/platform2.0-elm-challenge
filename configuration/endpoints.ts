const version = 1;
const apiBasePath = "https://api.thecatapi.com";

export const endpoints = {
    getAllCats: `${apiBasePath}/v${version}/images/search`,
    getImage: `${apiBasePath}/v${version}/images/`,
    favorite: `${apiBasePath}/v${version}/favourites/`
};