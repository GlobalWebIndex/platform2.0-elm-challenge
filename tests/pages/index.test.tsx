import { render, screen } from '@testing-library/react';
import { dummyIndex } from 'configuration/dummy';
import Index from 'pages/index';
import '@testing-library/jest-dom';

describe('Index Page', () => {
    it("renders correctly", () => {
        const { container } = render(
            <Index data={dummyIndex} />
        );

        expect(container).toMatchSnapshot();
    });

    it('Home Page shows nothing found when no data are returned', () => {
        render(
            <Index data={[]} />
        );

        expect(screen.getByTestId("no-cats")).toBeInTheDocument();
    });
});