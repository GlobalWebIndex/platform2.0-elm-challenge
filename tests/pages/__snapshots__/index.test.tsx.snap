// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page renders correctly 1`] = `
<div>
  <div
    class="container mx-auto md:my-12"
  >
    <div
      class="sm:border-b-[1px] sm:border-b-gray-300 sm:dark:border-b-gray-800 mb-12 px-4 py-4"
    >
      <h1
        class="text-3xl mb-4"
      >
        index:title
      </h1>
      <p
        class="max-w-[800px]"
      >
        index:description
      </p>
    </div>
    <div
      class="grid sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-0 md:gap-8"
    >
      <div
        class="aspect-square shadow hover:cursor-pointer w-full relative"
      >
        <div
          class="hover:scale-120 transition-transform absolute top-4 left-4 cursor-pointer z-10 false"
        >
          <svg
            aria-hidden="true"
            class="drop-shadow-lg text-white"
            fill="currentColor"
            height="32"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.645 20.91l-.007-.003-.022-.012a15.247 15.247 0 01-.383-.218 25.18 25.18 0 01-4.244-3.17C4.688 15.36 2.25 12.174 2.25 8.25 2.25 5.322 4.714 3 7.688 3A5.5 5.5 0 0112 5.052 5.5 5.5 0 0116.313 3c2.973 0 5.437 2.322 5.437 5.25 0 3.925-2.438 7.111-4.739 9.256a25.175 25.175 0 01-4.244 3.17 15.247 15.247 0 01-.383.219l-.022.012-.007.004-.003.001a.752.752 0 01-.704 0l-.003-.001z"
            />
          </svg>
        </div>
        <a
          href="/breed/7gn"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="7gn"
              class="hover:opacity-60 transition-opacity"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(https://cdn2.thecatapi.com/images/7gn.jpg);"
            />
            <noscript />
          </span>
        </a>
      </div>
      <div
        class="aspect-square shadow hover:cursor-pointer w-full relative"
      >
        <div
          class="hover:scale-120 transition-transform absolute top-4 left-4 cursor-pointer z-10 false"
        >
          <svg
            aria-hidden="true"
            class="drop-shadow-lg text-white"
            fill="currentColor"
            height="32"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.645 20.91l-.007-.003-.022-.012a15.247 15.247 0 01-.383-.218 25.18 25.18 0 01-4.244-3.17C4.688 15.36 2.25 12.174 2.25 8.25 2.25 5.322 4.714 3 7.688 3A5.5 5.5 0 0112 5.052 5.5 5.5 0 0116.313 3c2.973 0 5.437 2.322 5.437 5.25 0 3.925-2.438 7.111-4.739 9.256a25.175 25.175 0 01-4.244 3.17 15.247 15.247 0 01-.383.219l-.022.012-.007.004-.003.001a.752.752 0 01-.704 0l-.003-.001z"
            />
          </svg>
        </div>
        <a
          href="/breed/9sp"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="9sp"
              class="hover:opacity-60 transition-opacity"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(https://cdn2.thecatapi.com/images/9sp.jpg);"
            />
            <noscript />
          </span>
        </a>
      </div>
      <div
        class="aspect-square shadow hover:cursor-pointer w-full relative"
      >
        <div
          class="hover:scale-120 transition-transform absolute top-4 left-4 cursor-pointer z-10 false"
        >
          <svg
            aria-hidden="true"
            class="drop-shadow-lg text-white"
            fill="currentColor"
            height="32"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.645 20.91l-.007-.003-.022-.012a15.247 15.247 0 01-.383-.218 25.18 25.18 0 01-4.244-3.17C4.688 15.36 2.25 12.174 2.25 8.25 2.25 5.322 4.714 3 7.688 3A5.5 5.5 0 0112 5.052 5.5 5.5 0 0116.313 3c2.973 0 5.437 2.322 5.437 5.25 0 3.925-2.438 7.111-4.739 9.256a25.175 25.175 0 01-4.244 3.17 15.247 15.247 0 01-.383.219l-.022.012-.007.004-.003.001a.752.752 0 01-.704 0l-.003-.001z"
            />
          </svg>
        </div>
        <a
          href="/breed/a8k"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="a8k"
              class="hover:opacity-60 transition-opacity"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(https://cdn2.thecatapi.com/images/a8k.jpg);"
            />
            <noscript />
          </span>
        </a>
      </div>
      <div
        class="aspect-square shadow hover:cursor-pointer w-full relative"
      >
        <div
          class="hover:scale-120 transition-transform absolute top-4 left-4 cursor-pointer z-10 false"
        >
          <svg
            aria-hidden="true"
            class="drop-shadow-lg text-white"
            fill="currentColor"
            height="32"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.645 20.91l-.007-.003-.022-.012a15.247 15.247 0 01-.383-.218 25.18 25.18 0 01-4.244-3.17C4.688 15.36 2.25 12.174 2.25 8.25 2.25 5.322 4.714 3 7.688 3A5.5 5.5 0 0112 5.052 5.5 5.5 0 0116.313 3c2.973 0 5.437 2.322 5.437 5.25 0 3.925-2.438 7.111-4.739 9.256a25.175 25.175 0 01-4.244 3.17 15.247 15.247 0 01-.383.219l-.022.012-.007.004-.003.001a.752.752 0 01-.704 0l-.003-.001z"
            />
          </svg>
        </div>
        <a
          href="/breed/af8"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="af8"
              class="hover:opacity-60 transition-opacity"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(https://cdn2.thecatapi.com/images/af8.jpg);"
            />
            <noscript />
          </span>
        </a>
      </div>
      <div
        class="aspect-square shadow hover:cursor-pointer w-full relative"
      >
        <div
          class="hover:scale-120 transition-transform absolute top-4 left-4 cursor-pointer z-10 false"
        >
          <svg
            aria-hidden="true"
            class="drop-shadow-lg text-white"
            fill="currentColor"
            height="32"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.645 20.91l-.007-.003-.022-.012a15.247 15.247 0 01-.383-.218 25.18 25.18 0 01-4.244-3.17C4.688 15.36 2.25 12.174 2.25 8.25 2.25 5.322 4.714 3 7.688 3A5.5 5.5 0 0112 5.052 5.5 5.5 0 0116.313 3c2.973 0 5.437 2.322 5.437 5.25 0 3.925-2.438 7.111-4.739 9.256a25.175 25.175 0 01-4.244 3.17 15.247 15.247 0 01-.383.219l-.022.012-.007.004-.003.001a.752.752 0 01-.704 0l-.003-.001z"
            />
          </svg>
        </div>
        <a
          href="/breed/b0n"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="b0n"
              class="hover:opacity-60 transition-opacity"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(https://cdn2.thecatapi.com/images/b0n.jpg);"
            />
            <noscript />
          </span>
        </a>
      </div>
      <div
        class="aspect-square shadow hover:cursor-pointer w-full relative"
      >
        <div
          class="hover:scale-120 transition-transform absolute top-4 left-4 cursor-pointer z-10 false"
        >
          <svg
            aria-hidden="true"
            class="drop-shadow-lg text-white"
            fill="currentColor"
            height="32"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.645 20.91l-.007-.003-.022-.012a15.247 15.247 0 01-.383-.218 25.18 25.18 0 01-4.244-3.17C4.688 15.36 2.25 12.174 2.25 8.25 2.25 5.322 4.714 3 7.688 3A5.5 5.5 0 0112 5.052 5.5 5.5 0 0116.313 3c2.973 0 5.437 2.322 5.437 5.25 0 3.925-2.438 7.111-4.739 9.256a25.175 25.175 0 01-4.244 3.17 15.247 15.247 0 01-.383.219l-.022.012-.007.004-.003.001a.752.752 0 01-.704 0l-.003-.001z"
            />
          </svg>
        </div>
        <a
          href="/breed/bkc"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="bkc"
              class="hover:opacity-60 transition-opacity"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(https://cdn2.thecatapi.com/images/bkc.jpg);"
            />
            <noscript />
          </span>
        </a>
      </div>
      <div
        class="aspect-square shadow hover:cursor-pointer w-full relative"
      >
        <div
          class="hover:scale-120 transition-transform absolute top-4 left-4 cursor-pointer z-10 false"
        >
          <svg
            aria-hidden="true"
            class="drop-shadow-lg text-white"
            fill="currentColor"
            height="32"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.645 20.91l-.007-.003-.022-.012a15.247 15.247 0 01-.383-.218 25.18 25.18 0 01-4.244-3.17C4.688 15.36 2.25 12.174 2.25 8.25 2.25 5.322 4.714 3 7.688 3A5.5 5.5 0 0112 5.052 5.5 5.5 0 0116.313 3c2.973 0 5.437 2.322 5.437 5.25 0 3.925-2.438 7.111-4.739 9.256a25.175 25.175 0 01-4.244 3.17 15.247 15.247 0 01-.383.219l-.022.012-.007.004-.003.001a.752.752 0 01-.704 0l-.003-.001z"
            />
          </svg>
        </div>
        <a
          href="/breed/cfk"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="cfk"
              class="hover:opacity-60 transition-opacity"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(https://cdn2.thecatapi.com/images/cfk.jpg);"
            />
            <noscript />
          </span>
        </a>
      </div>
      <div
        class="aspect-square shadow hover:cursor-pointer w-full relative"
      >
        <div
          class="hover:scale-120 transition-transform absolute top-4 left-4 cursor-pointer z-10 false"
        >
          <svg
            aria-hidden="true"
            class="drop-shadow-lg text-white"
            fill="currentColor"
            height="32"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.645 20.91l-.007-.003-.022-.012a15.247 15.247 0 01-.383-.218 25.18 25.18 0 01-4.244-3.17C4.688 15.36 2.25 12.174 2.25 8.25 2.25 5.322 4.714 3 7.688 3A5.5 5.5 0 0112 5.052 5.5 5.5 0 0116.313 3c2.973 0 5.437 2.322 5.437 5.25 0 3.925-2.438 7.111-4.739 9.256a25.175 25.175 0 01-4.244 3.17 15.247 15.247 0 01-.383.219l-.022.012-.007.004-.003.001a.752.752 0 01-.704 0l-.003-.001z"
            />
          </svg>
        </div>
        <a
          href="/breed/MTUwNjcwMQ"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="MTUwNjcwMQ"
              class="hover:opacity-60 transition-opacity"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(https://cdn2.thecatapi.com/images/MTUwNjcwMQ.jpg);"
            />
            <noscript />
          </span>
        </a>
      </div>
      <div
        class="aspect-square shadow hover:cursor-pointer w-full relative"
      >
        <div
          class="hover:scale-120 transition-transform absolute top-4 left-4 cursor-pointer z-10 false"
        >
          <svg
            aria-hidden="true"
            class="drop-shadow-lg text-white"
            fill="currentColor"
            height="32"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.645 20.91l-.007-.003-.022-.012a15.247 15.247 0 01-.383-.218 25.18 25.18 0 01-4.244-3.17C4.688 15.36 2.25 12.174 2.25 8.25 2.25 5.322 4.714 3 7.688 3A5.5 5.5 0 0112 5.052 5.5 5.5 0 0116.313 3c2.973 0 5.437 2.322 5.437 5.25 0 3.925-2.438 7.111-4.739 9.256a25.175 25.175 0 01-4.244 3.17 15.247 15.247 0 01-.383.219l-.022.012-.007.004-.003.001a.752.752 0 01-.704 0l-.003-.001z"
            />
          </svg>
        </div>
        <a
          href="/breed/NVGe5-dNr"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="NVGe5-dNr"
              class="hover:opacity-60 transition-opacity"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(https://cdn2.thecatapi.com/images/NVGe5-dNr.jpg);"
            />
            <noscript />
          </span>
        </a>
      </div>
      <div
        class="aspect-square shadow hover:cursor-pointer w-full relative"
      >
        <div
          class="hover:scale-120 transition-transform absolute top-4 left-4 cursor-pointer z-10 false"
        >
          <svg
            aria-hidden="true"
            class="drop-shadow-lg text-white"
            fill="currentColor"
            height="32"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.645 20.91l-.007-.003-.022-.012a15.247 15.247 0 01-.383-.218 25.18 25.18 0 01-4.244-3.17C4.688 15.36 2.25 12.174 2.25 8.25 2.25 5.322 4.714 3 7.688 3A5.5 5.5 0 0112 5.052 5.5 5.5 0 0116.313 3c2.973 0 5.437 2.322 5.437 5.25 0 3.925-2.438 7.111-4.739 9.256a25.175 25.175 0 01-4.244 3.17 15.247 15.247 0 01-.383.219l-.022.012-.007.004-.003.001a.752.752 0 01-.704 0l-.003-.001z"
            />
          </svg>
        </div>
        <a
          href="/breed/Z6UtarTTe"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              alt="Z6UtarTTe"
              class="hover:opacity-60 transition-opacity"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(https://cdn2.thecatapi.com/images/Z6UtarTTe.png);"
            />
            <noscript />
          </span>
        </a>
      </div>
    </div>
  </div>
  <div
    class="fixed w-screen bottom-4 md:bottom-12 flex justify-center z-20"
  >
    <button
      class="disabled:opacity-0 disabled:pointer-events-none bg-gradient-to-br from-gray-200 to-gray-100 dark:from-gray-800 dark:to-gray-900 backdrop-blur-lg px-12 py-2 md:py-3 block "
      data-testid="button"
      type="button"
    >
      common:loadmore
    </button>
  </div>
</div>
`;
